.github/actions/test-env-setup/action.yml

```diff
@@ -0,0 +1,45 @@
+name: 'Set up CCBC dev environment'
+description: 'Sets up asdf, perl, and configures the cache'
+runs:
+  using: composite
+  steps:
+    - name: Install asdf
+      uses: asdf-vm/actions/setup@v1
+    - name: Cache asdf Postgres
+      uses: actions/cache@v2
+      env:
+        cache-name: asdf-postgres-cache
+      with:
+        path: |
+          /home/runner/.asdf
+        key: ${{ runner.os }}-postgres-14.1-${{ env.cache-name }}-0.3
+        restore-keys: |
+          ${{ runner.os }}-postgres-14.1-${{ env.cache-name }}-0.3
+    - name: Setup Postgres DB
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -yqq sqitch libdbd-pg-perl postgresql-client
+        asdf plugin-add postgres || true
+        asdf plugin-update --all
+        asdf install postgres 14.1
+        asdf reshim
+        pg_ctl start
+        createdb ${{ env.PGDATABASE }} || true
+        sqitch deploy
```

Comment:
With a `sqitch deploy` here, you're going to save the state of the database in your cache, which means that, because our cache key won't change when sqitch migrations change, it won't redeploy changes, any you'll have a stale schema, and , if we delete/rename a change, sqitch will exit with an error.

This action should only install the required apt packages, set env variables, and set up the caches, as it does in cas-cif: https://github.com/bcgov/cas-cif/blob/develop/.github/actions/dev-env-setup/action.yml

You should have a [separate action](https://github.com/bcgov/cas-cif/blob/develop/.github/actions/local-app-run/action.yaml) to deploy the migrations and start the app, ideally by running the docker containers that we're building instead of the apt-installed sqitch